src/rgw/rgw_rados.cc

```diff
@@ -6091,7 +6095,11 @@ int RGWRados::Bucket::UpdateIndex::complete(int64_t poolid, uint64_t epoch,
 
   int ret = get_bucket_shard(&bs);
   if (ret < 0) {
-    ldout(store->ctx(), 5) << "failed to get BucketShard object: ret=" << ret << dendl;
+    if (ret == -ENOENT) {
```

Comment:
less sure about this path;  what request are we executing, and what has happened in the current request to this point?